src/pokenav.c

```diff
@@ -0,0 +1,1198 @@
+        sub_81CC524,
+        sub_81CC670,
+    },
+    {
+        sub_81CCFD8,
+        sub_81CD070,
+        sub_81CDDD4,
+        sub_81CDE2C,
+        sub_81CDE64,
+        sub_81CD1C0,
+        sub_81CECA0,
+    },
+    {
+        sub_81CEF3C,
+        sub_81CEFDC,
+        sub_81CF330,
+        sub_81CF3A0,
+        sub_81CF3D0,
+        sub_81CEFF0,
+        sub_81CF3F8,
+    },
+    {
+        sub_81CD024,
+        sub_81CD070,
+        sub_81CDDD4,
+        sub_81CDE2C,
+        sub_81CDE64,
+        sub_81CD1C0,
+        sub_81CECA0,
+    },
+    {
+        sub_81CEF98,
+        sub_81CEFDC,
+        sub_81CF368,
+        sub_81CF3A0,
+        sub_81CF3D0,
+        sub_81CEFF0,
+        sub_81CF3F8,
+    },
+    {
+        sub_81CAAE8,
+        sub_81CAB24,
+        sub_81CB260,
+        sub_81CB29C,
+        sub_81CB2CC,
+        sub_81CAB38,
+        sub_81CB2E0,
+    },
+    {
+        sub_81CF9BC,
+        sub_81CFA34,
+        sub_81CFDD0,
+        sub_81CFE40,
+        sub_81CFE70,
+        sub_81CFA48,
+        sub_81CFE98,
+    },
+    {
+        sub_81D0450,
+        sub_81D04A0,
+        sub_81D0978,
+        sub_81D09B0,
+        sub_81D09E0,
+        sub_81D04B8,
+        sub_81D09F4,
+    },
+    {
+        sub_81CFA04,
+        sub_81CFA34,
+        sub_81CFE08,
+        sub_81CFE40,
+        sub_81CFE70,
+        sub_81CFA48,
+        sub_81CFE98,
+    },
+};
+
+const u16 gUnknown_0861F590[] = INCBIN_U16("graphics/pokenav/icon2.gbapal");
+const u32 gUnknown_0861F5B0[] = INCBIN_U32("graphics/pokenav/icon2.4bpp.lz");
+const u32 gUnknown_0861F994[] = INCBIN_U32("graphics/pokenav/icon2_unused.4bpp.lz");
+
+const struct BgTemplate gUnknown_0861FA04 =
+{
+    .bg = 0,
+    .charBaseIndex = 0,
+    .mapBaseIndex = 5,
+    .screenSize = 0,
+    .paletteMode = 0,
+    .priority = 0,
+    .baseTile = 0,
+};
+
+const struct WindowTemplate gUnknown_0861FA08[2] =
+{
+    {
+        .bg = 0,
+        .tilemapLeft = 1,
+        .tilemapTop = 0x16,
+        .width = 0x10,
+        .height = 0x2,
+        .paletteNum = 0,
+        .baseBlock = 0x36,
+    },
+    {
+        .bg = 0xFF,
+        .tilemapLeft = 0,
+        .tilemapTop = 0,
+        .width = 0,
+        .height = 0,
+        .paletteNum = 0,
+        .baseBlock = 0,
+    },
+};
+
+const u8 *const (MenuButtonReminders[12]) =
+{
+    gText_Navgear_ClearButtonList,
+    gText_NavgearMap_ZoomedOutButtons,
+    gText_NavgearMap_ZoomedInButtons,
+    gText_NavgearCondition_MonListButtons,
+    gText_NavgearCondition_MonStatusButtons,
+    gText_NavgearCondition_MarkingButtons,
+    gText_NavgearMatchCall_TrainerListButtons,
+    gText_NavgearMatchCall_CallMenuButtons,
+    gText_NavgearMatchCall_CheckTrainerButtons,
+    gText_NavgearRibbons_MonListButtons,
+    gText_NavgearRibbons_RibbonListButtons,
+    gText_NavgearRibbons_RibbonCheckButtons,
+};
+
+const u8 gMenuButtonReminderColor[4] =
+{
+    4, 1, 2, 0
+};
+
+const struct CompressedSpriteSheet gUnknown_0861FA4C[1] =
```

Comment:
This doesn't need to be an array

Reply:
But it was an array in gamefreak's code. It's accessed within a for loop that goes exactly one iteration. It's really silly.

Reply:
```C
    for (i = 0; i == 0; i++)
    {
        LoadCompressedSpriteSheet(&gUnknown_0861FA4C[i]);
    }
```
In `sub_81C7C28`.

Reply:
Interesting... Then leave it as an array of unspecified length (`gUnknown_0861FA4C[]`) and have that for loop check against `ARRAY_COUNT(gUnknown_0861FA4C)`.

Reply:
Alright.